packages/domains-table/src/status-popover/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { Gridicon, Popover } from '@automattic/components';
+import { useI18n } from '@wordpress/react-i18n';
+import classNames from 'classnames';
+import React, { useState, useRef } from 'react';
+import './style.scss';
+
+type StatusPopoverProps = {
+	children: React.ReactNode;
+	className?: string;
+};
+
+export const StatusPopover = ( { children, className }: StatusPopoverProps ) => {
+	const iconRef = useRef( null );
+	const [ showPopover, setShowPopover ] = useState( false );
+	const [ inPopover, setInPopover ] = useState( false );
+	const { __ } = useI18n();
+
+	const handleOnMouseEnterButton = () => {
+		setShowPopover( true );
+	};
+
+	const handleOnMouseLeave = () => {
+		setTimeout( () => {
+			if ( inPopover ) {
+				return;
+			}
+
+			setShowPopover( false );
+		}, 250 );
+	};
```

Comment:
I discovered that sometimes the popover would be stuck open. I'm not sure exactly how I did it with the mouse, but I guess it's a timing thing. Somehow `inPopover` behind left `true` when the timeout expires.

I tried to see if I could get a better result using the `useDebounce` hook (from the `use-debounce` package). I think I had better results by just tracking the hover state of each element and then debouncing the value of `inPopover || inButton`.

```tsx
	const [ inPopover, setInPopover ] = useState( false );
	const [ inButton, setInButton ] = useState( false );
	const [ showPopover ] = useDebounce( inPopover || inButton, 250 );

	const handleOnMouseEnterButton = () => {
		setInButton( true );
	};

	const handleOnMouseLeave = () => {
		setInButton( false );
	};

	const handleOnMouseEnterPopover = () => {
		setInPopover( true );
	};

	const handleOnMouseLeavePopover = () => {
		setInPopover( false );
	};
```